### PR TITLE
[KYV-665] Implemented middleware in monitoring-experience-api to make ip whitelist check for requests.

### PR DESCRIPTION
**Pull request by martinldev**
Implemented new middleware in monitoring-experience-api to enable endpoint ip whitelist access check for incoming requests.

### CODE REVIEW

File: packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import type {
+  AbstractController,
+  UnknownRequest
+} from '@verkkokauppa/core'
+import ipRangeCheck from 'ip-range-check'
+import { IpAddressValidationError } from '../errors'
+
+export const withIpWhitelistValidation = <
+    // eslint-disable-next-line prettier/prettier
```
Comment (severikupari1) [inline packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts:9]: Do not add these eslint disables

File: packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    TController extends abstract new (...args: any[]) => AbstractController
+    >(
+    Controller: TController
+): TController => {
+  abstract class IpAddressWhitelistValidationController extends Controller {
+    protected async validateRequest(req: UnknownRequest) {
+      const validIps = process.env.ROUTE_PUBLIC_MONITORING_IP_WHITELIST
+          ? process.env.ROUTE_PUBLIC_MONITORING_IP_WHITELIST.split(' ')
```
Comment (severikupari1) [inline packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts:17]: Should the split character be comma?

File: packages/monitoring-experience-api/src/routes.ts [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+import { withIpWhitelistValidation } from './middleware/ipWhitelistValidation'
 
 const router = Router()
 
-const elasticSearchHealthController = new ElasticSearchHealthController()
-const healthController = new Health()
+const elasticSearchHealthController = new (withIpWhitelistValidation(
+  ElasticSearchHealthController
+))()
+const healthController = new (withIpWhitelistValidation(Health))()
```
Comment (severikupari1) [inline packages/monitoring-experience-api/src/routes.ts:11]: Do not add whitelist validation to this health endpoint, only to elasticsearch health

**Review by severikupari1 — CHANGES_REQUESTED** [review]
Little questions

File: packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts (hunk 2 above)
Comment (martinldev) [reply packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts:17]: No, the IP:s are in the environment variable separated only with one space.

File: packages/monitoring-experience-api/src/routes.ts (hunk 3 above)
Comment (martinldev) [reply packages/monitoring-experience-api/src/routes.ts:11]: Sure, I will remove it from the health.

File: packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts (hunk 1 above)
Comment (martinldev) [reply packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts:9]: Yea, I'll delete this row.

File: packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+export const withIpWhitelistValidation = <
+  TController extends new (...args: any[]) => AbstractController
+>(
+  Controller: TController
+): TController => {
+  abstract class IpAddressWhitelistValidationController extends Controller {
+    protected async validateRequest(req: UnknownRequest) {
+      const validIps = process.env.ROUTE_PUBLIC_MONITORING_IP_WHITELIST
+        ? process.env.ROUTE_PUBLIC_MONITORING_IP_WHITELIST.split(' ')
```
Comment (severikupari1) [inline packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts:13]: Could you add some development value to this environment variable, to files
.env.docker
.env.local.docker

File: packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        : ['127.0.0.1']
+
+      const requestClientIp =
+        req.headers['x-forwarded-for'] || req.socket?.remoteAddress
+
+      if (!requestClientIp) {
+        throw new IpAddressValidationError('No IP found to validate')
+      }
+
+      let parsedClientIp
```
Comment (severikupari1) [inline packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts:23]: Could you add some comments what is happening in these splits, its hard to see why these are happenin.

Could you also write some kind of automation test and move this parsin functionality to own service?
Like UrlParseUtil etc

File: packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      if (Array.isArray(requestClientIp)) {
+        parsedClientIp = requestClientIp[0]
+      } else {
+        parsedClientIp = requestClientIp.split(', ')[0]
```
Comment (severikupari1) [inline packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts:27]: I dont understand this line, what does this do?

**Review by severikupari1 — APPROVED** [review]
good changes, thanks!

**Pull request merged**